app/src/main/java/uk/gov/mint/LoadHandler.java

```diff
@@ -13,25 +13,28 @@
 public class LoadHandler {
     private final CanonicalJsonMapper canonicalJsonMapper;
     private final EntriesUpdateDAO entriesUpdateDAO;
+    private final EntryValidator entryValidator;
 
-    public LoadHandler(EntriesUpdateDAO entriesUpdateDAO) {
+    public LoadHandler(EntriesUpdateDAO entriesUpdateDAO, EntryValidator entryValidator) {
         this.entriesUpdateDAO = entriesUpdateDAO;
+        this.entryValidator = entryValidator;
         this.canonicalJsonMapper = new CanonicalJsonMapper();
         entriesUpdateDAO.ensureTableExists();
     }
 
-    public void handle(String payload) throws Exception {
-        processEntries(payload.split("\n"));
+    public void handle(String registerName, String payload) {
+        processEntries(registerName, payload.split("\n"));
     }
 
-    private void processEntries(String[] entries) throws Exception {
+    private void processEntries(String registerName, String[] entries) {
         final List<byte[]> entriesAsBytes = Arrays.stream(entries)
                 .map(e -> {
                     try {
                         final JsonNode jsonNode = canonicalJsonMapper.readFromBytes(e.getBytes(StandardCharsets.UTF_8));
+                        entryValidator.validateEntry(registerName, jsonNode);
                         return canonicalJsonMapper.writeToBytes(hashedEntry(jsonNode));
-                    } catch (Exception e1) {
-                        throw new RuntimeException("Error parsing json entry: " + e, e1);
+                    } catch (Throwable throwable) {
+                        throw new RuntimeException(throwable);
```

Comment:
why was this changed? why would you want to wrap all Throwables in another RuntimeException?